src/util/real.h

```diff
@@ -0,0 +1,59 @@
+/*******************************************************************\
+
+Module: Real Numbers
+
+Author: Michael Tautschnig
+
+\*******************************************************************/
+
+#ifndef CPROVER_UTIL_REAL_H
+#define CPROVER_UTIL_REAL_H
+
+#include "mp_arith.h"
+#include "std_expr.h"
+
+class realt
+{
+protected:
+  mp_integer integral, fractional;
```

Comment:
Representing real numbers like rationals will at the very least irritate. Please add a comment why this is the right choice.

Reply:
Hmm, which approach would you have chosen instead?

Reply:
E.g., you could have represented the algebraic numbers via the root of a polynomial; I'd be curious what the various SMT solvers output as models.

Reply:
I'll try to address this (even when I'd argue that the current PR already addresses a number of CBMC invariant failures that the included regression tests cause when attempted without the code changes).

Reply:
No need to implement this right now; my suggestion is to add a comment, then merge this PR.

Reply:
I've added a comment and a KNOWNBUG regression test (amounting to the example that @polgreen gave).

Reply:
I have now added a further commit that removes `realt` and introduces `algebraic_numbert` while also fixing that `KNOWNBUG` test introduced in the prior commit.